packages/app/src/cli/utilities/app/http-reverse-proxy.ts

```diff
@@ -66,18 +66,20 @@ export async function runConcurrentHTTPProcessesAndPathForwardTraffic({
     }),
   )
 
-  const availablePort = portNumber ?? (await getAvailableTCPPort())
-
   outputDebug(outputContent`
-Starting reverse HTTP proxy on port ${outputToken.raw(availablePort.toString())}
+Starting reverse HTTP proxy on port ${outputToken.raw(portNumber.toString())}
 Routing traffic rules:
 ${outputToken.json(JSON.stringify(rules))}
 `)
 
   const proxy = httpProxy.createProxy()
   const server = http.createServer(function (req, res) {
     const target = match(rules, req)
-    if (target) return proxy.web(req, res, {target})
+    if (target) {
+      return proxy.web(req, res, {target}, (err) => {
+        outputWarn(`Error forwarding web request: ${err}`)
```

Comment:
For ruby apps, does this mean we are almost always going to show this warning while the app is starting up?

Reply:
Yup!

Reply:
What about having state scoped to the target that indicates whether it's running or not. Then we can use it to:
- Output it as a `debug` (when we pass `--verbose`) if the server is starting.
- Output it  as a warning always if the server was started and all of a sudden we started getting errors.

Reply:
makes sense, i'd also like to get it rendered with the other processes inside `renderConcurrent`